### PR TITLE
Update README

### LINKED ISSUE #105: Update readme
The link to Substitution Cipher is broken in readme and newer inclusions are not present.

### PR DESCRIPTION
**Pull request by ivaibhavjindal**
Closes #105

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,29 +34,29 @@ This is a library of various algorithms that are used in competitive programming
...
-[Z Algorithm](Algorithms/String_Hashing/z-algorithm.cpp)
-[Manacher Algorithm](Algorithms/String_Hashing/manacher.cpp)
+[Rabin Karp Algorithm](Algorithms/String_Hashing/rabin_karp.cpp) <br/>
+[Count Substring](Algorithms/String_Hashing/count_substr.cpp) <br/>
+[Z Algorithm](Algorithms/String_Hashing/z-algorithm.cpp) <br/>
+[Manacher Algorithm](Algorithms/String_Hashing/manacher.cpp) <br/>
 
 ### Backtracking Algorithms
-[Hamiltonian Cycle](Algorithms/Backtracking/Hamiltonian%20Cycle.cpp)
-[M Coloring Problem](Algorithms/Backtracking/M%20Coloring%20Problem.cpp)
```
Comment (Bhavnaharitsa) [inline README.md]: no need of these they are already properly aligned

File: README.md [hunk 2]
```diff
@@ -34,29 +34,29 @@ This is a library of various algorithms that are used in competitive programming
...
+[Bubble Sort](Algorithms/Sorting/bubble_sort.cpp) <br/>
+[Selection Sort](Algorithms/Sorting/selection_sort.cpp) <br/>
 [Radix Sort](Algorithms/Sorting/radix%20sort.cpp)
 
 ### String Hashing
-[Rabin Karp Algorithm](Algorithms/String_Hashing/rabin_karp.cpp)
-[Count Substring](Algorithms/String_Hashing/count_substr.cpp)
-[Z Algorithm](Algorithms/String_Hashing/z-algorithm.cpp)
-[Manacher Algorithm](Algorithms/String_Hashing/manacher.cpp)
+[Rabin Karp Algorithm](Algorithms/String_Hashing/rabin_karp.cpp) <br/>
```
Comment (Bhavnaharitsa) [inline README.md:50]: no need of these they are already properly aligned

File: README.md [hunk 3]
```diff
@@ -34,29 +34,29 @@ This is a library of various algorithms that are used in competitive programming
...
+[Binary Search](Algorithms/Searching/binarysearch.cpp) <br/>
 [Jump Search](Algorithms/Searching/jumpsearch.cpp)
 
 ### Sorting Algorithms 
-[Merge Sort](Algorithms/Sorting/merge_sort.cpp)
-[Heap Sort](Algorithms/Sorting/heap_sort.cpp)
-[Insertion Sort](Algorithms/Sorting/insertion_sort.cpp)
-[Bubble Sort](Algorithms/Sorting/bubble_sort.cpp)
-[Selection Sort](Algorithms/Sorting/selection_sort.cpp)
+[Merge Sort](Algorithms/Sorting/merge_sort.cpp) <br/>
```
Comment (Bhavnaharitsa) [inline README.md:42]: no need of these they are already properly aligned

File: README.md [hunk 4]
```diff
@@ -34,29 +34,29 @@ This is a library of various algorithms that are used in competitive programming
 [Kadane Algorithm](Kadane/kadane.cpp)
 
 ### Searching Algorithms
-[Linear Search](Algorithms/Searching/Linear_Search.cpp)
-[Binary Search](Algorithms/Searching/binarysearch.cpp)
+[Linear Search](Algorithms/Searching/Linear_Search.cpp) <br/>
```
Comment (Bhavnaharitsa) [inline README.md:37]: no need of these they are already properly aligned

File: README.md [hunk 5]
```diff
@@ -10,21 +10,21 @@ This is a library of various algorithms that are used in competitive programming
 [Circular Queue](Algorithms/Circular_Queue/CircularQueue.cpp)
 
 ### Graphs
-[Breadth First Search](Algorithms/Graphs/breadth_first_search.cpp)
-[Depth First Search](Algorithms/Graphs/DFS_BFS_in_Graphs.cpp)
-[Kruskal](Algorithms/Graphs/Kruskal's%20Algorithm.cpp)
-[Dijkstra](Algorithms/Graphs/Dijkstra's%20Algorithm.cpp)
+[Breadth First Search](Algorithms/Graphs/breadth_first_search.cpp) <br/>
```
Comment (Bhavnaharitsa) [inline README.md:13]: no need of these changes at all

File: README.md [hunk 6]
```diff
@@ -4,35 +4,63 @@ This is a library of various algorithms that are used in competitive programming
...
+[Bubble Sort](Algorithms/Sorting/bubble_sort.cpp)
 [Selection Sort](Algorithms/Sorting/selection_sort.cpp)
+[Radix Sort](Algorithms/Sorting/radix%20sort.cpp)
 
 ### String Hashing
-[Rabin Karp Algorithm](Algorithms/String_Hashing/rabin_karp.cpp)<br/>
-[Count_Substring](Algorithms/String_Hashing/count_substr.cpp) <br/>
-[Z Algorithm](Algorithms/String_Hashing/z-algorithm.cpp)<br/>
-[Manacher Algorithm](Algorithms/String_Hashing/manacher.cpp) <br/>
+[Rabin Karp Algorithm](Algorithms/String_Hashing/rabin_karp.cpp)
```
Comment (Bhavnaharitsa) [inline README.md]: just removing the _ in Count Substring would do
Compare in rich text

**Comment by ivaibhavjindal** [comment]
Sorry but I didn't understood the changes requested.
I get the fact that they were properly aligned but some had extra space or underscores or cpp in Headings which I removed to make it more consistent. The backtracking Algo links pointed to a specific repo so I made it relative just like other links. And the underscore in the Count substring has also been removed. You can see the updated readme at https://github.com/ivaibhavjindal/nwoc_algorithms/tree/read

**Comment by Bhavnaharitsa** [comment]
I meant even without the </br> the texts will under each other so unnecessarily we don't need that

**Comment by Bhavnaharitsa** [comment]
> Sorry but I didn't understood the changes requested.
> I get the fact that they were properly aligned but some had extra space or underscores or cpp in Headings which I removed to make it more consistent. The backtracking Algo links pointed to a specific repo so I made it relative just like other links. And the underscore in the Count substring has also been removed. You can see the updated readme at https://github.com/ivaibhavjindal/nwoc_algorithms/tree/read

The `</br>` changes that you did was unnecessary and the count_substring is still the same unchanged

**Comment by ivaibhavjindal** [comment]
I have made a new PR #121 with the required changes. Please review

**Pull request closed without merging**